server/src/main/java/io/druid/server/http/DatasourcesResource.java

```diff
@@ -143,14 +143,22 @@ public Response getTheDataSource(
       @QueryParam("full") final String full
   )
   {
-    ImmutableDruidDataSource dataSource = getDataSource(dataSourceName);
+    final ImmutableDruidDataSource dataSource = getDataSource(dataSourceName);
 
     if (dataSource == null) {
       return Response.noContent().build();
     }
 
     if (full != null) {
-      return Response.ok(dataSource).build();
+      final Map<String, Object> fullResult = ImmutableMap.of(
```

Comment:
Instead of this, how about having `@JsonProperty` on the `getSegments()` method, and having the `@JsonCreator` constructor accept `segments` and make the map out of it? It should be equivalent but cleaner -- usually we try to handle serde through annotations.